AndroidTool/Scripts/startRecordingForSerial.sh

```diff
@@ -21,27 +21,40 @@ then
     # Get resolution if no custom res was specified
     if [[ ! $width ]]
     then
-        width=`"$adb" -s $serial shell dumpsys display | grep mDisplayWidth | awk -F '=' '{ print $2 }' | tr -d '\r\n'`
+        width=`"$adb" -s $serial shell dumpsys display | awk -F '=' '/mDisplayWidth/ { print $2 }' | tr -d '\r\n'`
     fi
     if [[ ! $height ]]
     then
-        height=`"$adb" -s $serial shell dumpsys display | grep mDisplayHeight | awk -F '=' '{ print $2 }' | tr -d '\r\n'`
+        height=`"$adb" -s $serial shell dumpsys display | awk -F '=' '/mDisplayHeight/ { print $2 }' | tr -d '\r\n'`
     fi
     sizeopt=""
     # Put a --size option only if both params are available
     if [[ $width && $height ]]
     then
-        sizeopt=${width}x${height}
+        sizeopt="--size ${height}x${width}"
     fi
 
-    "$adb" -s $serial shell screenrecord --size $sizeopt --o raw-frames /sdcard/screencapture.raw
+    "$adb" -s $serial shell screenrecord $sizeopt --o raw-frames /sdcard/screencapture.raw
 else
     echo "Recording from phone..."
-    orientation=$("$adb" -s $serial shell dumpsys input | grep 'SurfaceOrientation' | awk '{ print $2 }')
+    
+    orientation=$("$adb" -s $serial shell dumpsys input | awk '/SurfaceOrientation/ { print $2 }')
+
+    sizeopt=""
     if [[ "${orientation//[$'\t\r\n ']}" != "0" ]]
     then
-        "$adb" -s $serial shell screenrecord --bit-rate $bitrate --verbose --size $height"x"$width /sdcard/capture.mp4 # > $1/reclog.txt
+        sizeopt="--size ${height}x${width}"
     else
-        "$adb" -s $serial shell screenrecord --bit-rate $bitrate --verbose --size $width"x"$height /sdcard/capture.mp4 # > $1/reclog.txt
+        sizeopt="--size ${width}x${height}"
+    fi
+    
+    "$adb" -s $serial shell screenrecord --verbose --bit-rate $bitrate ${sizeopt} /sdcard/capture.mp4  # > $1/reclog.txt
+    
+    ## [bugre] UMI Max with Android 7, is a 1920x1080 device, but recording setting this resolution
+    ## results in error. Must be: 1920x1088 or let it record without resolution defined.
+    ## Maybe this error also afects some other devices. 
+    if [ "$ret" != "0" ]; then
+        echo "Trying again, but now without setting the resolution."
+        "$adb" -s $serial shell screenrecord --verbose --bit-rate $bitrate /sdcard/capture.mp4  # > $1/reclog.txt
     fi
-fi
+fi
```

Comment:
Hi, 

No idea where this 'symbol' after 'fi' came from. 
![image](https://user-images.githubusercontent.com/6900672/58434424-f1643400-8091-11e9-94e5-7f0b8cf6aebc.png)



If it is real, please consider changing it before merging.
I've checked my code, with visual studio and even `od -c ..` , and i'm unable to find anything after 'fi' on that line..

![image](https://user-images.githubusercontent.com/6900672/58434341-86b2f880-8091-11e9-907f-17c5d90b8f37.png)

Reply:
You probably need to add a new line after 60.